lib/hope-tabs.js

```diff
@@ -0,0 +1,722 @@
+/*global fxosComponent, HopeTab, scheduler */
+(function() {
+
+var debug = 1 ? (...args) => console.log('[hope-tabs]', ...args) : () => {};
+
+const BAR_HEIGHT = HopeTab.BAR_HEIGHT;
+const PREVIEW_HEIGHT = 88;
+const GUTTER_HEIGHT = 8;
+const TAB_HEIGHT = BAR_HEIGHT + PREVIEW_HEIGHT + GUTTER_HEIGHT;
+
+fxosComponent.register('hope-tabs', {
+  created() {
+    var shadow = this.setupShadowRoot();
+
+    this.els = {
+      inner: shadow.querySelector('.inner'),
+      container: shadow.querySelector('.container'),
+      grippy: shadow.querySelector('.grippy'),
+      add: shadow.querySelector('.add'),
+      tabs: Array.from(this.querySelectorAll('li'))
+    };
+
+    this.onScroll = this.onScroll.bind(this);
+    this.selecting = false;
+    this.opening = false;
+    this.closing = false;
+
+    // Events
+    scheduler.attachDirect(this.els.inner, 'scroll', this.onScroll);
+    on(this.els.container, 'close-clicked', e => this.onClose(e));
+    on(this.els.container, 'click', e => this.onClick(e));
+    on(this.els.add, 'click', e => this.onAddClick(e));
+
+    this.setupSnapping();
+
+    // Layout when decendants are ready
+    setTimeout(() => this.layout());
+  },
+
+  onClick(e) {
```

Comment:
we need to add an early return case here when `!this.inTabsView`, otherwise taping the chrome yields a very weird (yet smooth!) transition when the tabs view is closed.

eventually the chrome will eat those taps probably but we're not there yet